File: app/src/main/java/com/example/fruitsdiary/util/DateUtils.java
package com.example.fruitsdiary.util;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public final class DateUtils {

    private static final String SERVER_DATE_PATTERN = "yyyy-MM-dd";

    private static final DateFormat SERVER_DATE_FORMAT = new SimpleDateFormat(SERVER_DATE_PATTERN);

    private static final DateFormat APP_DATE_FORMAT = DateFormat.getDateInstance(DateFormat.LONG);

    private DateUtils() {
    }

    public static String convertServerDateToAppDate(String serverDate) {
        Date date = getServerDate(serverDate);
        return getAppStringDate(date);
    }

    public static String getCurrentAppDate() {
        Date date = getCurrentDate();
        return getAppStringDate(date);
    }

    public static String getCurrentServerDate() {
        Date date = getCurrentDate();
        return SERVER_DATE_FORMAT.format(date);

    }

    private static Date getCurrentDate() {
        return Calendar.getInstance().getTime();
    }

    public static Date getServerDate(String serverDate) {
        try {
            return SERVER_DATE_FORMAT.parse(serverDate);
        } catch (ParseException e) {
            return new Date();
        }
    }

    public static String getAppStringDate(Date date){
        return APP_DATE_FORMAT.format(date);
    }
}

File: app/src/main/java/com/example/fruitsdiary/util/DateUtils.kt
package com.example.fruitsdiary.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    private const val SERVER_DATE_PATTERN = "yyyy-MM-dd"

    private val SERVER_DATE_FORMAT = SimpleDateFormat(SERVER_DATE_PATTERN)

    private val APP_DATE_FORMAT = SimpleDateFormat("MMMM dd, yyyy", Locale.US)

     @JvmStatic fun getCurrentAppDate(): String {
            val date = getCurrentDate()
            return getAppStringDate(date)
        }

    @JvmStatic fun getCurrentServerDate(): String {
            val date = getCurrentDate()
            return SERVER_DATE_FORMAT.format(date)
        }

    private fun getCurrentDate(): Date  = Calendar.getInstance().time

    @JvmStatic fun convertServerDateToAppDate(serverDate: String): String {
        val date = getServerDate(serverDate)
        return getAppStringDate(date)
    }

    private fun getServerDate(serverDate: String): Date {
        return try {
            SERVER_DATE_FORMAT.parse(serverDate)
        } catch (e: ParseException) {
            Date()
        }
    }

    private fun getAppStringDate(date: Date): String {
        return APP_DATE_FORMAT.format(date)
    }
}
